Promote subscription: Add setting to toggle Subscription modal

lupus2k opened the pull request:
Related to https://github.com/Automattic/verbum/issues/34

## Proposed Changes

* Add an option to enable/disable the subscription modal on Verbum on the `Subscriptions` section of `/settings/newsletter/` page

PT: pb5gDS-3tB-p2

![image](https://github.com/Automattic/wp-calypso/assets/402286/967f9afa-1a42-484d-8604-83b4cb68f349)


## Testing Instructions

* Go to `/settings/newsletter/` using a Simple site
* You should see the toggle.
* Try changing the site language
* Try using an Atomic site

lupus2k committed:
Promote subscription: Add setting to toggle Subscription modal

lsl reviewed (approved):
Looks ok, does this need a backend change to go with it? Currently it's not saving the toggled value.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9726767

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @lupus2k for including a screenshot in the description! This is really helpful for our translators.

provenself commented:
![Screenshot 2023-11-22 at 3 11 04 PM](https://github.com/Automattic/wp-calypso/assets/88352513/7f3588d9-aedf-45d5-aa99-7caf841d0133)

Editorial checking in. Phrasing looks good! However, we should remove the periods at the end of these two options, as they're the only ones with a period on the entire modal.

lupus2k commented:
> Looks ok, does this need a backend change to go with it? Currently it's not saving the toggled value.

Sorry, I forgot to add the Jetpack PR to the description 🤦 
https://github.com/Automattic/jetpack/pull/34258

lupus2k commented:
> Editorial checking in. Phrasing looks good! However, we should remove the periods at the end of these two options, as they're the only ones with a period on the entire modal.

Sure, I'll remove the periods.
Thanks for your quick review!

lupus2k committed:
Show only for English sites & removes periods from some copies

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9726767).

lupus2k committed:
Add jetpack_ prefix to the option

crisbusquets commented:
Hi!

Sorry if I'm too late, but:
- Can we remove the period at the end? No other option has it: `Display subscription suggestion after comment`
- I wonder if we should say `newsletter subscription` instead of just subscription. cc @simison

lupus2k commented:
> * Can we remove the period at the end? No other option has it: `Display subscription suggestion after comment`
> * I wonder if we should say `newsletter subscription` instead of just subscription. cc @simison

Hey @crisbusquets, I already removed the period and updated the screenshot on the PR.

No problem on copy change since we plan to release to all locales later

lupus2k committed:
Move toggle to settings/newsletter
